tensorflow/core/BUILD

```diff
@@ -780,6 +780,24 @@ cc_library(
     alwayslink = 1,
 )
 
+cc_library(
+    name = "training_op_helpers",
+    srcs = [
+        "//tensorflow/core/framework:training_op_helpers.cc",
+    ],
+    hdrs = [
+        "//tensorflow/core/framework:training_op_helpers.h",
+    ],
+    visibility = ["//visibility:public"],
```

Comment:
Not public visibility.

We also have many google-internal users of the library in kernels, so we should leave a forwarding header file or this PR will not be successfully merged.

Reply:
Hey @alextp, I'm not the original author of this PR. I've just checked it out due to it being needed in TensorFlow Addons (original: #29703). I fixed some of the issues during a local compile and created the PR. 

I don't fully understand what you mentioned, could I maybe get a bit more context? More specifically, is there an example in TensorFlow repo or any other that uses `bazel` and makes use of forwarding headers?

Reply:
1. Remove the line saying visibility:public; use at most something like //tensorflow:internal so it doesn't become a free-for-all.

2. Leave a file in tensorflow/core/kernels/training_op_helpers.h which just `#include`s the file in core/framework; this way preexisting code which imports the file in kernels can continue to build while we refactor it to get the library from the right place.